src/linker/configuration.py

```diff
@@ -49,6 +49,10 @@ def _validate(self) -> None:
                 f"Container engine '{self.container_engine}' is not supported."
             )
 
+        # Validate input data against constraints
+        for input_file in self.input_data:
+            input_file_validator(input_file)
```

Comment:
we could do this in pipeline.py and call self.config.input_data, but i figured might as well do it the first time we validate the config.

Reply:
I'm torn. My gut feeling is it doesn't belong here b/c this is the Config object and we should just be validating that. But if I squint enough, part of the config object is the list of input data to the pipeline and so maybe it makes sense to actually validate those here? idk, will be interesting to see what others think.

Reply:
yeah, your feelings are basically the same as mine!

Reply:
I took a shot at putting this in the pipeline instead--I had some difficulty properly testing that the log messages get parsed properly with the existing test functions; might be worth doing in a separate ticket.